### PR TITLE
[fix]: #2571: Better Kura Inspector UX.

### PR DESCRIPTION
**Pull request by SamHSmith**
Signed-off-by: Sam H. Smith <sam.henning.smith@protonmail.com>

### Description of the Change

All of Victor's suggestions have been implemented.

### CODE REVIEW

File: tools/kura_inspector/src/main.rs [hunk 1]
```diff
@@ -67,11 +67,33 @@ fn main() {
...
+                _internal_path_buf_for_modified_block_store_path.pop();
+                &_internal_path_buf_for_modified_block_store_path
+            } else {
+                block_store_path
+            }
+        }
+        None => block_store_path,
+    };
+
     let block_store = StdFileBlockStore::new(block_store_path);
```
Comment (Erigara) [inline tools/kura_inspector/src/main.rs:86]: I think this peace of code is a good place for `Cow`, we ether return reference as is or modify.

```suggestion
    let mut block_store_path: std::borrow::Cow<'_, Path> = block_store_path.into();

    if let Some(os_str_file_name) = block_store_path.file_name() {
        let file_name_str = os_str_file_name.to_str().unwrap_or("");
        if file_name_str == "blocks.data" || file_name_str == "blocks.index" {
            block_store_path.to_mut().pop();
        }
    }

    let block_store = StdFileBlockStore::new(&block_store_path);
```

**Review by Erigara — DISMISSED** [review]
LGTM

**Review by 6r1d — APPROVED** [review]
LGTM

**Review by 6r1d — APPROVED** [review]

File: tools/kura_inspector/src/main.rs [hunk 2]
```diff
@@ -67,11 +67,26 @@ fn main() {
...
+    let block_store = StdFileBlockStore::new(&block_store_path);
 
     let index_count = block_store
         .read_index_count()
         .expect("Failed to read index count from block store {block_store_path:?}.");
+
+    if index_count == 0 {
+        println!("The block store is empty.");
+        return;
+    }
```
Comment (Erigara) [inline tools/kura_inspector/src/main.rs:88]: Small thing i just noticed: `assert!` became redundant.

Comment (SamHSmith) [reply tools/kura_inspector/src/main.rs:88]: I think it's fine. The assert was never ment as UX it was ment to protect the code after from failing.

**Review by Erigara — APPROVED** [review]

**Pull request merged**